### PR TITLE
Make Address and Location polymorphic

### PR DESCRIPTION
**Pull request by reggieb**
This will allow:

- `enrollment.location` to store grid reference and easting/northing
- `contact.address`
- `parner.address`
- `address.location`

### CODE REVIEW

File: app/models/flood_risk_engine/enrollment.rb [hunk 1]
```diff
@@ -20,6 +19,8 @@ class Enrollment < ActiveRecord::Base
     # The Correspondence Contact Details related to this Application a.k.a Main Contact
     belongs_to :correspondence_contact, class_name: "Contact"
 
+    has_one :location, as: :locatable, dependent: :restrict_with_exception
```
Comment (autotelik) [inline app/models/flood_risk_engine/enrollment.rb]: what is this ? the name doesn't really tell us what business need this full fills ... if this is the location where exemption activity takes place then I think `site_address` more meaningful or activity_location ...  I suspect that each Exemption maybe linked to a site so may not even be needed ... future prrofs against multiple exemptions/sites to start of use something like  EnrollmentExemption

```
class EnrollmentExemption < ActiveRecord::Base

    belongs_to :enrollment,
    belongs_to :exemption
    belongs_to :site_address,
 
```

Comment (reggieb) [inline app/models/flood_risk_engine/enrollment.rb]: At the moment, the only requirement is that the location that the exemptions will refer to have a grid reference and easting/northing. All exemptions must refer to a single location (that part of the spec. see: https://eaflood.atlassian.net/browse/RIP-60). However, `exemption_location` may be a better name.

Comment (autotelik) [inline app/models/flood_risk_engine/enrollment.rb]: ok. I just now we did exactly what you are doing in WEX, with site address on Enrollment and then had to change it later on 

We know bridges, and other structures likely to come along plus already have     `has_many :enrollment_exemptions` and `FloodRiskEngine::EnrollmentExemption`  .. so seems pretty easy to delegate activity_location to  the EnrollmentExemption .. but :ship:

**Comment by reggieb** [comment]
I've changed `enrollment.location` as `enrollment.exemption_location`. Hopefully having locations as polymorphic it should be fairly easy to move them to the exemption (or other) object as necessary.

**Comment by reggieb** [comment]
Rebased

**Comment by autotelik** [comment]
Ok. shame not to just bite the bullet now, since we have the relationship to the Exemption in a separate table anyway-  but the location intrinsic to that relationship is not in that table

**Pull request merged**